app.py

```diff
@@ -107,14 +207,132 @@ def do_logout(request):
 
 @view_config(route_name="question", renderer='templates/questionpage.jinja2')
 def question(request):
-    return {}
+    if request.authenticated_userid:
+        user = User.get_by_username(request.authenticated_userid)
+        if request.method == "POST":
+            answer = request.params.get("answer")
+            question = Question.get_question_by_id(
+                request.params.get("question_id")
+            )
+            if (answer not in [None, ''] and question.id not in
+               [sub.question_id for sub in Submission.get_all_for_user(user)]):
+
+                Submission.new(
+                    user=user,
+                    question=question,
+                    answer=answer
+                )
+        questions = Question.all()
+        if questions:
+            submissions = Submission.get_all_for_user(user)
+            l = []
+            for q in questions:
+                if q.id not in [s.question_id for s in submissions]:
+                    l.append(q)
+            if l:
+                question = l[randint(0, len(l) - 1)]
+                x, u, y = make_data(question, user)
+                if x != [] and y != []:
+                    prediction = int(round(guess(x, u, y)))
```

Comment:
if `guess` were robust to being passed an empty list for `x` or `y`, then you could simplify lines 234-238 to this:

```python
prediction = int(round(guess(make_data(question, user))))
```